### ISSUE TITLE
False positive: unicorn/prefer-top-level-await triggered by z.catch usage (Zod Mini)

### ISSUE THREAD

**Issue by J3m5** [sindresorhus/eslint-plugin-unicorn#2684]
## Describe the bug

The rule `unicorn/prefer-top-level-await` reports a false positive when using `z.catch` from Zod Mini (https://github.com/colinhacks/zod). Example code:

```js
const paramsSchema = z.catch(
  z.object({
    q: z.optional(z.string()),
    page: z._default(z.coerce.number(), 1),
    structure: z.optional(z.string()),
    license: z.optional(z.coerce.boolean()),
    SNS: z.optional(z.coerce.boolean()),
  }),
  {
    q: undefined,
    page: 1,
    structure: undefined,
    license: undefined,
    SNS: undefined,
  }
);
```

This triggers the error:

```
Prefer top-level await over using a promise chain. (unicorn/prefer-top-level-await)
```

However, `z.catch` is not a promise chain or async code, but a schema utility from Zod. This is a false positive.

## To Reproduce
1. Use Zod Mini's `z.catch` in a file with `eslint-plugin-unicorn` enabled.
2. Observe the false positive error from `unicorn/prefer-top-level-await`.

## Expected behavior
No error should be reported for this usage, as it is not related to promises or async code.

## Environment
- ESLint version: (any)
- eslint-plugin-unicorn version: (latest)
- Node.js version: (any)
- Zod version: 4.x (Mini)

## Additional context
This is blocking usage of Zod Mini's schema fallback feature in codebases using this rule.

---
Repro from user workspace, see also: https://github.com/colinhacks/zod
